### ISSUE TITLE
Plot options dialog blocks cli

### ISSUE THREAD

**Issue by EltayebAhmed** [mantidproject/mslice#71]
The plot options dialog is a modal dialog. The fact you can not interact with the parent plot window unit the dialog is closed is to be expected. However until the plot dialog options is dismissed the user will not be able to interact with parent plot, the parent command line and the main GUI as well. This behaviour is very counter intuitive and must be changed.

To reproduce:
Open a plot option dialog. Try to use the command-line. It will be unresponsive.

**Comment by mducle** [comment]
@martyngigg I had a go with using `.setWindowModality(Qt.WindowModal)` instead of the default  `Qt.ApplicationModal` which blocks everything to do with the application. However, when I tested it out, I found that whilst it will not block the main GUI window, it still blocks the IPython commandline. The other possible option, `Qt.NonModal` allows the CLI to be used, but also does not block the parent figure window... not sure which is the better option. I guess can set it to `Qt.NonModal` and make our own code to ignore inputs to just the parent figure window... but this is a bit hack-ish.
